ethdb/pebble/pebble.go

```diff
@@ -361,6 +362,13 @@ func (d *Database) Stat(property string) (string, error) {
 // is treated as a key after all keys in the data store. If both is nil then it
 // will compact entire data store.
 func (d *Database) Compact(start []byte, limit []byte) error {
+	// There is no special flag to represent the end of key range
+	// in pebble(nil in leveldb). Use an ugly hack to construct a
+	// large key to represent it.
```

Comment:
Perhaps add a note that any prefixed database entry will be smaller than this due to the prefix; as for trie nodes we need the 32 byte 0xff because there might be a shared prefix starting with a number of 0xff-s, so 32 ensures than only a hash collision could touch it.